### DISCUSSION TITLE
6-Button Sega Genesis Controller

### DISCUSSION THREAD

**Discussion by CodeDrivenCode** [EmulatorJS/EmulatorJS#557] [General]
Im looking to make a virtual controller for the 6-Button sega controller that has X Y Z on top and A B C on the bottom.  I have modified the emu-main.js to include the following code but its not working.

} else if ('segaMD' === getSystem(_this.system, true)) {
                    _0x48ee51 = {
                        0: 'B',
 			1: 'Y',
                        3: 'START',
                        4: 'UP',
                        5: 'DOWN',
                        6: 'LEFT',
                        7: 'RIGHT',
                        8: 'A',
			9: 'X',
                        10: 'C',
                        11: 'Z',
                        24: _this.localization('QUICK SAVE STATE'),
                        25: _this.localization('QUICK LOAD STATE'),
                        26: _this.localization('CHANGE STATE SLOT')
                    };

I have matching button settings here (the input_value of the buttons matches the number in the code above).

} else if (getSystem(_this.system, true) === 'segaMD') {

All the buttons work, but they are mismatched. For example pushing C returns a value of X and so on. 

Any help would be appreciated.

**Comment by ethanaobrien** [comment]
You need to change the input value. 

Read the docs: https://emulatorjs.org/docs4devs/Virutal%20Gamepad%20Settings.html

**Reply by CodeDrivenCode** [reply]
I figured it out

**Comment by CodeDrivenCode** [comment]
I have tried all kinds of input_values without luck. Could you be more specific?
